Library/Homebrew/test/string_spec.rb

```diff
@@ -0,0 +1,49 @@
+require "extend/string"
+
+describe String do
+  describe "#undent" do
+    it "removes leading whitespace, taking the first line as reference" do
+      expect(
+        <<-EOS.undent
+          hi
+..........my friend over
+            there
+        EOS
+      ).to eq("hi\n..........my friend over\n  there\n")
+    end
+
+    it "removes nothing if the text is not indented" do
+      expect(
+        <<-EOS.undent
+hi
+I'm not indented
+        EOS
+      ).to eq("hi\nI'm not indented\n")
+    end
+
+    it "can be nested" do
+      nested_string = <<-EOS.undent
+        goodbye
+      EOS
+
+      expect(
+        <<-EOS.undent
+          hello
+          #{nested_string}
+        EOS
+      ).to eq("hello\ngoodbye\n\n")
+    end
+  end
+end
+
+describe StringInreplaceExtension do
+  subject { string.extend(described_class) }
+  let(:string) { "foobar" }
+
+  describe "#errors" do
+    it "returns errors from #sub!" do
```

Comment:
it returns errors from `#sub!` <ins>when the string is not found</ins>.

Maybe we should have a test for the positive case too? Although since there currently isn't one I don't really mind either way since it's kind of out of this PR's scope.